### PR TITLE
Wallet API: add support for wallet creation from hardware device

### PR DESCRIPTION
**Pull request by stoffu**
CC: @cslashm

### CODE REVIEW

**Comment by stoffu** [comment]
@cslashm 
Because I wanted simplewallet to print the error message when the parsing fails. Error reporting is not included in `tools:: parse_subaddress_lookahead`, and the wallet API silently falls back to the default value when the parsing fails.

File: src/wallet/api/wallet_manager.cpp [hunk 1]
```diff
@@ -114,6 +114,27 @@ Wallet *WalletManagerImpl::createWalletFromKeys(const std::string &path,
...
+                                                  const std::string &subaddressLookahead)
+{
+    WalletImpl * wallet = new WalletImpl(nettype);
+    if(restoreHeight > 0){
+        wallet->setRefreshFromBlockHeight(restoreHeight);
+    }
+    auto lookahead = tools::parse_subaddress_lookahead(subaddressLookahead);
+    if (!lookahead)
+    {
+        lookahead = tools::parse_subaddress_lookahead("5:20");
```
Comment (moneromooo-monero) [inline src/wallet/api/wallet_manager.cpp]: Maybe std::make_pair(5,20) ?

File: src/wallet/api/wallet_manager.cpp [hunk 2]
```diff
@@ -114,6 +114,27 @@ Wallet *WalletManagerImpl::createWalletFromKeys(const std::string &path,
...
+    }
+    wallet->setSubaddressLookahead(lookahead->first, lookahead->second);
```
Comment (moneromooo-monero) [inline src/wallet/api/wallet_manager.cpp]: Though with this, i'll override the wallet2 settings even when not specified. The defaults should be in wallet2 so other programs benefit from a good value.

Comment (stoffu) [reply src/wallet/api/wallet_manager.cpp]: OK, please see the first commit.

File: src/wallet/wallet2.cpp [hunk 3]
```diff
@@ -3269,6 +3269,12 @@ void wallet2::restore(const std::string& wallet_, const epee::wipeable_string& p
   cryptonote::block b;
   generate_genesis(b);
   m_blockchain.push_back(get_block_hash(b));
+  if (m_subaddress_lookahead_major == SUBADDRESS_LOOKAHEAD_MAJOR && m_subaddress_lookahead_major == SUBADDRESS_LOOKAHEAD_MINOR)
```
Comment (moneromooo-monero) [inline src/wallet/wallet2.cpp]: The second major should be minor

Comment (stoffu) [reply src/wallet/wallet2.cpp]: Ouch :P

**Comment by luigi1111** [comment]
@stoffu please rebase.

**Comment by stoffu** [comment]
Rebased

**Pull request merged**